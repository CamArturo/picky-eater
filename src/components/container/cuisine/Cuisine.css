.cuisine-header {
  background: url("../../../assets/food-200.jpg");
  height: 200px; }
  .cuisine-header h2 {
    font-family: Signika Negative, serif;
    font-style: normal;
    font-weight: bold;
    line-height: 56px;
    font-size: 36px;
    letter-spacing: 0.04em;
    text-align: center;
    padding: 48px 0 0 0;
    margin: 0;
    color: #fff;
    text-shadow: 3px 3px 4px rgba(0, 0, 0, 0.7); }

.cuisine-container {
  display: flex;
  flex-wrap: wrap;
  height: 100vh; }
  .cuisine-container .return-restaurants-container {
    width: 100%;
    height: 100px;
    margin: 1rem 0;
    display: flex;
    justify-content: center;
    align-items: center; }
    .cuisine-container .return-restaurants-container .return-restaurants {
      width: auto;
      height: 40px;
      font-family: Catamaran, serif;
      font-weight: bold;
      font-size: 18px;
      line-height: 37px;
      letter-spacing: 2px;
      color: #fff;
      background-color: #0956B2;
      border: none;
      margin: 0; }
      .cuisine-container .return-restaurants-container .return-restaurants:hover {
        background-color: #1468CC;
        color: #fff;
        cursor: pointer; }
  .cuisine-container .cuisine-btn {
    width: 10%;
    min-height: 100px;
    height: 100px;
    display: flex;
    justify-content: center;
    align-items: center;
    border: 1px solid #cecece; }
    .cuisine-container .cuisine-btn:hover {
      background-color: #cecece; }
    .cuisine-container .cuisine-btn h3 {
      pointer-events: none;
      font-family: Catamaran, serif;
      font-style: italic;
      font-weight: normal;
      line-height: 150%;
      letter-spacing: 0.04em;
      color: #000; }

.loading-gif {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
  width: 100%; }

.no-locations {
  height: 100vh;
  margin: 1rem auto 0; }

/* Large Devices, Wide Screens */
/* Medium Devices, Desktops */
/* Small Devices, Tablets */
/* Extra Small Devices, Phones */
/* Custom, iPhone Retina */

/*# sourceMappingURL=Cuisine.css.map */
