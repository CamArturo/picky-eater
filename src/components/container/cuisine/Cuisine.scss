.cuisine-header {
  background: url("../../../assets/food-200.jpg");
  height: 200px;
  h2 {
    font-family: Signika Negative, serif;
    font-style: normal;
    font-weight: bold;
    line-height: 56px;
    font-size: 36px;
    letter-spacing: 0.04em;
    text-align: center;
    padding: 48px 0 0 0;
    margin: 0;

    color: #fff;
    text-shadow: 3px 3px 4px rgba(0, 0, 0, 0.7);
  }
}

.cuisine-container {
  display: flex;
  flex-wrap: wrap;
  .cuisine-btn-container {
    width: 100%;
    display: -ms-flexbox;
    height: 100vh;
    //display: flex;
    //-ms-flex-wrap: wrap;
    //flex-wrap: wrap;
  }
  .cuisine-info {
    width: 100%;
    p {
      margin: 2rem auto 1rem;
      text-align: center;
    }
  }
  .return-restaurants-container {
    width: 100%;
    height: 100px;
    margin: 1rem 0;
    display: flex;
    justify-content: center;
    align-items: center;
    .return-restaurants {
      width: auto;
      height: 40px;
      font-family: Catamaran, serif;
      font-weight: bold;
      font-size: 18px;
      line-height: 37px;
      letter-spacing: 2px;
      color: #fff;
      background-color: #0956b2;
      border: none;
      margin: 0;
      &:hover {
        background-color: #1468cc;
        color: #fff;
        cursor: pointer;
      }
    }
  }
  .cuisine-btn {
    //width: 240px;
    width: 10%;
    min-height: 100px;
    height: 100px;
    //height: 200px;
    //display: flex;
    //justify-content: center;
    //align-items: center;
    float: left;
    border: 1px solid #cecece;
    &:hover {
      background-color: #cecece;
    }
    h3 {
      //width: 124px;
      //height: 51px;
      pointer-events: none;

      font-family: Catamaran, serif;
      font-style: italic;
      font-weight: normal;
      line-height: 150%;
      //font-size: 2rem;
      letter-spacing: 0.04em;

      color: #000;
    }
  }
}

.loading-gif {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
  width: 100%;
}

.no-locations {
  height: 100vh;
  margin: 1rem auto 0;
}

/* Large Devices, Wide Screens */
@media only screen and (max-width: 1200px) {

}

/* Medium Devices, Desktops */
@media only screen and (max-width: 992px) {

}

/* Small Devices, Tablets */
@media only screen and (max-width: 768px) {

}

/* Extra Small Devices, Phones */
@media only screen and (max-width: 480px) {
  .cuisine-container {
    .cuisine-btn {
      width: 100%;
    }
  }
}

/* Custom, iPhone Retina */
@media only screen and (max-width: 320px) {

}