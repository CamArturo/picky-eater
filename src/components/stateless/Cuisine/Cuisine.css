.cuisine-container-header {
  background: url("../../../assets/food-200.jpg");
  height: 200px; }
  .cuisine-container-header h2 {
    font-family: Signika Negative, serif;
    font-style: normal;
    font-weight: bold;
    line-height: 56px;
    font-size: 36px;
    letter-spacing: 0.04em;
    text-align: center;
    padding: 48px 0 0 0;
    margin: 0;
    color: #fff;
    text-shadow: 3px 3px 4px rgba(0, 0, 0, 0.7); }

.cuisine-container {
  display: flex;
  flex-wrap: wrap; }
  .cuisine-container .cuisine-card {
    width: 240px;
    height: 200px;
    display: flex;
    justify-content: center;
    align-items: center; }

/* Large Devices, Wide Screens */
/* Medium Devices, Desktops */
/* Small Devices, Tablets */
/* Extra Small Devices, Phones */
/* Custom, iPhone Retina */

/*# sourceMappingURL=Cuisine.css.map */
