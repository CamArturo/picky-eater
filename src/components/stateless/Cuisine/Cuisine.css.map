{
"version": 3,
"mappings": "AAAA,eAAgB;EACd,UAAU,EAAE,mCAAmC;EAC/C,MAAM,EAAE,KAAK;EACb,kBAAG;IACD,WAAW,EAAE,uBAAuB;IACpC,UAAU,EAAE,MAAM;IAClB,WAAW,EAAE,IAAI;IACjB,WAAW,EAAE,IAAI;IACjB,SAAS,EAAE,IAAI;IACf,cAAc,EAAE,MAAM;IACtB,UAAU,EAAE,MAAM;IAClB,OAAO,EAAE,UAAU;IACnB,MAAM,EAAE,CAAC;IAET,KAAK,EAAE,IAAI;IACX,WAAW,EAAE,8BAA8B;;AAI/C,kBAAmB;EACjB,OAAO,EAAE,IAAI;EACb,SAAS,EAAE,IAAI;EACf,gCAAc;IACZ,KAAK,EAAE,KAAK;IACZ,MAAM,EAAE,KAAK;IACb,OAAO,EAAE,IAAI;IACb,eAAe,EAAE,MAAM;IACvB,WAAW,EAAE,MAAM;IACnB,MAAM,EAAE,iBAAiB;IACzB,mCAAG;MACD,KAAK,EAAE,KAAK;MACZ,MAAM,EAAE,IAAI;MAEZ,WAAW,EAAE,gBAAgB;MAC7B,UAAU,EAAE,MAAM;MAClB,WAAW,EAAE,MAAM;MACnB,WAAW,EAAE,IAAI;MACjB,SAAS,EAAE,IAAI;MACf,cAAc,EAAE,MAAM;MAEtB,KAAK,EAAE,IAAI;;AAKjB,iCAAiC;AAKjC,8BAA8B;AAK9B,4BAA4B;AAK5B,iCAAiC;AAKjC,2BAA2B",
"sources": ["Cuisine.scss"],
"names": [],
"file": "Cuisine.css"
}