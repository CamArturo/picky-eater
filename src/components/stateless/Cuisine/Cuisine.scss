.cuisine-header {
  background: url("../../../assets/food-200.jpg");
  height: 200px;
  h2 {
    font-family: Signika Negative, serif;
    font-style: normal;
    font-weight: bold;
    line-height: 56px;
    font-size: 36px;
    letter-spacing: 0.04em;
    text-align: center;
    padding: 48px 0 0 0;
    margin: 0;

    color: #fff;
    text-shadow: 3px 3px 4px rgba(0, 0, 0, 0.7);
  }
}

.cuisine-container {
  display: flex;
  flex-wrap: wrap;
  .cuisine-card {
    width: 240px;
    height: 200px;
    display: flex;
    justify-content: center;
    align-items: center;
    border: 1px solid #cecece;
    h3 {
      width: 124px;
      height: 51px;

      font-family: Catamaran, serif;
      font-style: italic;
      font-weight: normal;
      line-height: 150%;
      font-size: 2rem;
      letter-spacing: 0.04em;

      color: #000;
    }
  }
}

/* Large Devices, Wide Screens */
@media only screen and (max-width: 1200px) {

}

/* Medium Devices, Desktops */
@media only screen and (max-width: 992px) {

}

/* Small Devices, Tablets */
@media only screen and (max-width: 768px) {

}

/* Extra Small Devices, Phones */
@media only screen and (max-width: 480px) {

}

/* Custom, iPhone Retina */
@media only screen and (max-width: 320px) {

}