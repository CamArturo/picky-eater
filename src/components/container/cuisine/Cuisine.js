import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Loading from '../../stateless/loading/Loading';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import { getCuisines } from '../../../api/getCuisines';

import { sendCuisinesToStore } from '../../../actions/cuisines';
import './Cuisine.css';

export class Cuisine extends Component {
  constructor (props) {
    super(props);
  }

  componentDidMount () {
    if (this.props.chosenCityID) {
      this.storeCuisines(this.props.chosenCityID);
    }
  }

  displayNoLocations = () => {
    return (
      <section className="no-locations">
        <p>No Locations Chosen</p>
      </section>
    );
  };

  storeCuisines = async (chosenCityID) => {
    const cuisines = await getCuisines(chosenCityID);
    this.props.loadCuisines(cuisines.cuisines);
  };
//
// {
//   cuisine: {
//     cuisine_id: 1,
//     cuisine_name: 'American'
//   }
// }

  displayCuisines = () => {
    return this.props.availableCuisines.map((cuisine, index) => (
      <button
        onClick={(event) => this.filterAvailableCuisines(event.target.name)}
        className="cuisine-btn"
        key={`key${index}`}
        name={cuisine.cuisine.cuisine_name}
      >
        <h3>{cuisine.cuisine.cuisine_name}</h3>
      </button>
    ));
  };

  filterAvailableCuisines = (name) => {
    const availableCuisines = this.props.availableCuisines;

    const remainingCuisines = availableCuisines.filter((cuisineObj, index) => {
      return cuisineObj.cuisine.cuisine_name !== name;
    });
    this.props.loadCuisines(remainingCuisines);
  };

  render () {
    return (
      <div>
        <section className="cuisine-header">
          <h2>What do you NOT want to eat?</h2>
        </section>
        <section className="cuisine-container">
          {
            Object.keys(this.props.chosenCity).length === 0 &&
            this.displayNoLocations()
          }
          {
            this.props.availableCuisines.length > 0 &&
            this.displayCuisines()
          }
          {
            Object.keys(this.props.chosenCity).length > 0 && this.props.availableCuisines.length === 0 &&
            <Loading />
          }
        </section>
      </div>
    );
  }
}

Cuisine.propTypes = {
  getCuisines: PropTypes.func.isRequired,
  sendCuisinesToStore: PropTypes.func.isRequired,
  loadCuisines: PropTypes.func.isRequired,
  availableCuisines: PropTypes.array.isRequired,
  chosenCityID: PropTypes.number.isRequired,
  chosenCity: PropTypes.object.isRequired
};

export const mapStateToProps = (state) => ({
  availableCuisines: state.availableCuisines,
  chosenCityID: state.chosenCity.id,
  chosenCity: state.chosenCity
});

export const mapDispatchToProps = (dispatch) => ({
  loadCuisines: (cities) => dispatch(sendCuisinesToStore(cities))
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Cuisine));
