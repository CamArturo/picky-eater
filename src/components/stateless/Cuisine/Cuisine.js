import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Loading from '../Loading/Loading';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import { getCuisines } from '../../../api/Api';
import { sendCuisinesToStore } from '../../../actions/cuisines';
import './Cuisine.css';

class Cuisine extends Component {
  constructor (props) {
    super(props);
  }

  componentDidMount () {
    if (this.props.chosenCityID) {
      this.storeCuisines(this.props.chosenCityID);
    }
  }

  displayNoLocations = () => {
    return (
      <section className="no-locations">
        <p>No Locations Chosen</p>
      </section>
    );
  };

  storeCuisines = async (chosenCityID) => {
    const cuisines = await getCuisines(chosenCityID);
    this.props.loadCuisines(cuisines.cuisines);
  };

  displayCuisines = () => {
    return this.props.availableCuisines.map((cuisine, index) => (
      <button className="cuisine-btn" key={`key${index}`}>
        <h3>{cuisine.cuisine.cuisine_name}</h3>
      </button>
    ));
  };

  render () {
    return (
      <div>
        <section className="cuisine-header">
          <h2>What do you NOT want to eat?</h2>
        </section>
        <section className="cuisine-container">
          {
            this.props.chosenCity &&
            this.displayNoLocations()
          }
          {
            this.props.availableCuisines.length > 0 ?
              this.displayCuisines() : <Loading/>
          }
        </section>
      </div>
    );
  }
}

// Cuisine.propTypes = {};

const mapStateToProps = (state) => ({
  availableCuisines: state.availableCuisines,
  chosenCityID: state.chosenCity.id
});

const mapDispatchToProps = (dispatch) => ({
  loadCuisines: (cities) => dispatch(sendCuisinesToStore(cities))
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Cuisine));
