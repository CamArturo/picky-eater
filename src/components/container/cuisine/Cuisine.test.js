import React from 'react';
import { Cuisine } from './Cuisine';
import { shallow, mount } from 'enzyme';
import * as mock from '../../../mockData';
// import { getCuisines } from '../../../api/getCuisines'
jest.mock('../../../api/getCuisines');

describe('Cuisine page tests', () => {

  let wrapper;
  let storeCuisines;
  let chosenCityID;
  let getCuisines;
  let sendCuisinesToStore;
  let loadCuisines;
  let availableCuisines;
  let chosenCity;

  beforeEach(() => {
    storeCuisines = jest.fn();
    sendCuisinesToStore = jest.fn();
    loadCuisines = jest.fn();
    getCuisines = jest.fn();
    availableCuisines = mock.availableCuisines;
    chosenCityID = 305;
    chosenCity = mock.chosenCity;

    wrapper = shallow(<Cuisine
        storeCuisines={storeCuisines}
        chosenCityID={chosenCityID}
        getCuisines={getCuisines}
        sendCuisinesToStore={sendCuisinesToStore}
        loadCuisines={loadCuisines}
        availableCuisines={availableCuisines}
        chosenCity={chosenCity}
      />,
      {
        disableLifecycleMethods: false
      });
  });

  it('Should match the snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });

  it('it calls storeCuisines on componentDidMount when chosenCityID has value', async () => {

    expect(loadCuisines).toHaveBeenCalled();
  });

  // it.skip('should ', () => {
  //   // jest.spyOn - finds a method and just listening for it. 'method'
  //   const spy = jest.spyOn(wrapper.instance(), 'storeCuisines');
  //   await wrapper.instance().componentDidMount();
  //   expect(spy).toHaveBeenCalled();
  // });

});