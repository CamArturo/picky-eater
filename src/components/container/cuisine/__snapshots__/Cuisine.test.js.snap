// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cuisine page tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cuisine
    availableCuisines={
        Array [
            Object {
              "cuisine": Object {
                "cuisine_id": 6,
                "cuisine_name": "Afghani",
              },
            },
            Object {
              "cuisine": Object {
                "cuisine_id": 152,
                "cuisine_name": "African",
              },
            },
            Object {
              "cuisine": Object {
                "cuisine_id": 1,
                "cuisine_name": "American",
              },
            },
          ]
    }
    chosenCity={
        Object {
            "city": "Newton, IL",
            "id": 5172,
          }
    }
    chosenCityID={5172}
    getCuisines={[Function]}
    loadCuisines={[Function]}
    loadRestaurants={[Function]}
    sendCuisinesToStore={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="cuisine-header"
>
          <h2>
                    What do you NOT want to eat?
          </h2>
</section>,
        <section
          className="cuisine-container"
>
          <div
                    className="cuisine-info"
          >
                    <p>
                              Click to Remove the cuisines you do not want
                    </p>
                    <section
                              className="return-restaurants-container"
                    >
                              <NavLink
                                        activeClassName="active"
                                        ariaCurrent="true"
                                        to="/restaurants"
                              >
                                        <button
                                                  className="return-restaurants"
                                                  onClick={[Function]}
                                        >
                                                  Show me these Restaurants
                                        </button>
                              </NavLink>
                    </section>
          </div>
          <section
                    className="cuisine-btn-container"
          >
                    <button
                              className="cuisine-btn"
                              name="Afghani"
                              onClick={[Function]}
                              title="Click to Remove"
                    >
                              <h3>
                                        Afghani
                              </h3>
                    </button>
                    <button
                              className="cuisine-btn"
                              name="African"
                              onClick={[Function]}
                              title="Click to Remove"
                    >
                              <h3>
                                        African
                              </h3>
                    </button>
                    <button
                              className="cuisine-btn"
                              name="American"
                              onClick={[Function]}
                              title="Click to Remove"
                    >
                              <h3>
                                        American
                              </h3>
                    </button>
          </section>
</section>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h2>
            What do you NOT want to eat?
</h2>,
          "className": "cuisine-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "What do you NOT want to eat?",
          },
          "ref": null,
          "rendered": "What do you NOT want to eat?",
          "type": "h2",
        },
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="cuisine-info"
>
              <p>
                            Click to Remove the cuisines you do not want
              </p>
              <section
                            className="return-restaurants-container"
              >
                            <NavLink
                                          activeClassName="active"
                                          ariaCurrent="true"
                                          to="/restaurants"
                            >
                                          <button
                                                        className="return-restaurants"
                                                        onClick={[Function]}
                                          >
                                                        Show me these Restaurants
                                          </button>
                            </NavLink>
              </section>
</div>,
            false,
            <section
              className="cuisine-btn-container"
>
              <button
                            className="cuisine-btn"
                            name="Afghani"
                            onClick={[Function]}
                            title="Click to Remove"
              >
                            <h3>
                                          Afghani
                            </h3>
              </button>
              <button
                            className="cuisine-btn"
                            name="African"
                            onClick={[Function]}
                            title="Click to Remove"
              >
                            <h3>
                                          African
                            </h3>
              </button>
              <button
                            className="cuisine-btn"
                            name="American"
                            onClick={[Function]}
                            title="Click to Remove"
              >
                            <h3>
                                          American
                            </h3>
              </button>
</section>,
            false,
          ],
          "className": "cuisine-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Click to Remove the cuisines you do not want
</p>,
                <section
                  className="return-restaurants-container"
>
                  <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    to="/restaurants"
                  >
                                    <button
                                                      className="return-restaurants"
                                                      onClick={[Function]}
                                    >
                                                      Show me these Restaurants
                                    </button>
                  </NavLink>
</section>,
              ],
              "className": "cuisine-info",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Click to Remove the cuisines you do not want",
                },
                "ref": null,
                "rendered": "Click to Remove the cuisines you do not want",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/restaurants"
>
                    <button
                                        className="return-restaurants"
                                        onClick={[Function]}
                    >
                                        Show me these Restaurants
                    </button>
</NavLink>,
                  "className": "return-restaurants-container",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": "restaurant-nav",
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "ariaCurrent": "true",
                    "children": <button
                      className="return-restaurants"
                      onClick={[Function]}
>
                      Show me these Restaurants
</button>,
                    "to": "/restaurants",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Show me these Restaurants",
                      "className": "return-restaurants",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Show me these Restaurants",
                    "type": "button",
                  },
                  "type": [Function],
                },
                "type": "section",
              },
            ],
            "type": "div",
          },
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="cuisine-btn"
                  name="Afghani"
                  onClick={[Function]}
                  title="Click to Remove"
>
                  <h3>
                                    Afghani
                  </h3>
</button>,
                <button
                  className="cuisine-btn"
                  name="African"
                  onClick={[Function]}
                  title="Click to Remove"
>
                  <h3>
                                    African
                  </h3>
</button>,
                <button
                  className="cuisine-btn"
                  name="American"
                  onClick={[Function]}
                  title="Click to Remove"
>
                  <h3>
                                    American
                  </h3>
</button>,
              ],
              "className": "cuisine-btn-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "key0",
                "nodeType": "host",
                "props": Object {
                  "children": <h3>
                    Afghani
</h3>,
                  "className": "cuisine-btn",
                  "name": "Afghani",
                  "onClick": [Function],
                  "title": "Click to Remove",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Afghani",
                  },
                  "ref": null,
                  "rendered": "Afghani",
                  "type": "h3",
                },
                "type": "button",
              },
              Object {
                "instance": null,
                "key": "key1",
                "nodeType": "host",
                "props": Object {
                  "children": <h3>
                    African
</h3>,
                  "className": "cuisine-btn",
                  "name": "African",
                  "onClick": [Function],
                  "title": "Click to Remove",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "African",
                  },
                  "ref": null,
                  "rendered": "African",
                  "type": "h3",
                },
                "type": "button",
              },
              Object {
                "instance": null,
                "key": "key2",
                "nodeType": "host",
                "props": Object {
                  "children": <h3>
                    American
</h3>,
                  "className": "cuisine-btn",
                  "name": "American",
                  "onClick": [Function],
                  "title": "Click to Remove",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "American",
                  },
                  "ref": null,
                  "rendered": "American",
                  "type": "h3",
                },
                "type": "button",
              },
            ],
            "type": "section",
          },
          false,
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="cuisine-header"
>
            <h2>
                        What do you NOT want to eat?
            </h2>
</section>,
          <section
            className="cuisine-container"
>
            <div
                        className="cuisine-info"
            >
                        <p>
                                    Click to Remove the cuisines you do not want
                        </p>
                        <section
                                    className="return-restaurants-container"
                        >
                                    <NavLink
                                                activeClassName="active"
                                                ariaCurrent="true"
                                                to="/restaurants"
                                    >
                                                <button
                                                            className="return-restaurants"
                                                            onClick={[Function]}
                                                >
                                                            Show me these Restaurants
                                                </button>
                                    </NavLink>
                        </section>
            </div>
            <section
                        className="cuisine-btn-container"
            >
                        <button
                                    className="cuisine-btn"
                                    name="Afghani"
                                    onClick={[Function]}
                                    title="Click to Remove"
                        >
                                    <h3>
                                                Afghani
                                    </h3>
                        </button>
                        <button
                                    className="cuisine-btn"
                                    name="African"
                                    onClick={[Function]}
                                    title="Click to Remove"
                        >
                                    <h3>
                                                African
                                    </h3>
                        </button>
                        <button
                                    className="cuisine-btn"
                                    name="American"
                                    onClick={[Function]}
                                    title="Click to Remove"
                        >
                                    <h3>
                                                American
                                    </h3>
                        </button>
            </section>
</section>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h2>
              What do you NOT want to eat?
</h2>,
            "className": "cuisine-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "What do you NOT want to eat?",
            },
            "ref": null,
            "rendered": "What do you NOT want to eat?",
            "type": "h2",
          },
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="cuisine-info"
>
                <p>
                                Click to Remove the cuisines you do not want
                </p>
                <section
                                className="return-restaurants-container"
                >
                                <NavLink
                                                activeClassName="active"
                                                ariaCurrent="true"
                                                to="/restaurants"
                                >
                                                <button
                                                                className="return-restaurants"
                                                                onClick={[Function]}
                                                >
                                                                Show me these Restaurants
                                                </button>
                                </NavLink>
                </section>
</div>,
              false,
              <section
                className="cuisine-btn-container"
>
                <button
                                className="cuisine-btn"
                                name="Afghani"
                                onClick={[Function]}
                                title="Click to Remove"
                >
                                <h3>
                                                Afghani
                                </h3>
                </button>
                <button
                                className="cuisine-btn"
                                name="African"
                                onClick={[Function]}
                                title="Click to Remove"
                >
                                <h3>
                                                African
                                </h3>
                </button>
                <button
                                className="cuisine-btn"
                                name="American"
                                onClick={[Function]}
                                title="Click to Remove"
                >
                                <h3>
                                                American
                                </h3>
                </button>
</section>,
              false,
            ],
            "className": "cuisine-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Click to Remove the cuisines you do not want
</p>,
                  <section
                    className="return-restaurants-container"
>
                    <NavLink
                                        activeClassName="active"
                                        ariaCurrent="true"
                                        to="/restaurants"
                    >
                                        <button
                                                            className="return-restaurants"
                                                            onClick={[Function]}
                                        >
                                                            Show me these Restaurants
                                        </button>
                    </NavLink>
</section>,
                ],
                "className": "cuisine-info",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Click to Remove the cuisines you do not want",
                  },
                  "ref": null,
                  "rendered": "Click to Remove the cuisines you do not want",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <NavLink
                      activeClassName="active"
                      ariaCurrent="true"
                      to="/restaurants"
>
                      <button
                                            className="return-restaurants"
                                            onClick={[Function]}
                      >
                                            Show me these Restaurants
                      </button>
</NavLink>,
                    "className": "return-restaurants-container",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": "restaurant-nav",
                    "nodeType": "function",
                    "props": Object {
                      "activeClassName": "active",
                      "ariaCurrent": "true",
                      "children": <button
                        className="return-restaurants"
                        onClick={[Function]}
>
                        Show me these Restaurants
</button>,
                      "to": "/restaurants",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Show me these Restaurants",
                        "className": "return-restaurants",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "Show me these Restaurants",
                      "type": "button",
                    },
                    "type": [Function],
                  },
                  "type": "section",
                },
              ],
              "type": "div",
            },
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="cuisine-btn"
                    name="Afghani"
                    onClick={[Function]}
                    title="Click to Remove"
>
                    <h3>
                                        Afghani
                    </h3>
</button>,
                  <button
                    className="cuisine-btn"
                    name="African"
                    onClick={[Function]}
                    title="Click to Remove"
>
                    <h3>
                                        African
                    </h3>
</button>,
                  <button
                    className="cuisine-btn"
                    name="American"
                    onClick={[Function]}
                    title="Click to Remove"
>
                    <h3>
                                        American
                    </h3>
</button>,
                ],
                "className": "cuisine-btn-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "key0",
                  "nodeType": "host",
                  "props": Object {
                    "children": <h3>
                      Afghani
</h3>,
                    "className": "cuisine-btn",
                    "name": "Afghani",
                    "onClick": [Function],
                    "title": "Click to Remove",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Afghani",
                    },
                    "ref": null,
                    "rendered": "Afghani",
                    "type": "h3",
                  },
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": "key1",
                  "nodeType": "host",
                  "props": Object {
                    "children": <h3>
                      African
</h3>,
                    "className": "cuisine-btn",
                    "name": "African",
                    "onClick": [Function],
                    "title": "Click to Remove",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "African",
                    },
                    "ref": null,
                    "rendered": "African",
                    "type": "h3",
                  },
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": "key2",
                  "nodeType": "host",
                  "props": Object {
                    "children": <h3>
                      American
</h3>,
                    "className": "cuisine-btn",
                    "name": "American",
                    "onClick": [Function],
                    "title": "Click to Remove",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "American",
                    },
                    "ref": null,
                    "rendered": "American",
                    "type": "h3",
                  },
                  "type": "button",
                },
              ],
              "type": "section",
            },
            false,
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
